macOSUpgrade.sh

```diff
@@ -419,7 +395,9 @@ else
 fi
 
 ##Load LaunchAgent
-if [[ ${fvStatus} == "FileVault is On." ]] && [[ ${currentUser} != "root" ]]; then
+if [[ ${fvStatus} == "FileVault is On." ]] && \
+   [[ ${currentUser} != "root" ]] && \
+   [[ ${cancelFVAuthReboot} -eq 0 ]] ; then
     userID=$( /usr/bin/id -u "${currentUser}" )
     /bin/launchctl bootstrap gui/"${userID}" /Library/LaunchAgents/com.apple.install.osinstallersetupd.plist
```

Comment:
@kc9wwh 
We have the issues of https://github.com/kc9wwh/macOSUpgrade/issues/74 and https://github.com/kc9wwh/macOSUpgrade/issues/56.

I thought it would be good to decide the direction of this project!

## Background
When FileVault2 and APFS is on and user is not admin, the admin dialog is displayed.

## Policy Ideas
**Idea 1.** If a non-administrative user runs this script, it should uniformly not auth reboot. (We don't use the agent of osinstallersetupd.)
**Idea 2.** Temporarily add administrative privileges as https://github.com/kc9wwh/macOSUpgrade/pull/86 in this script.
**Idea 3.** According to this PR, depending on the arguments reboot will bring up the admin dialog.

Reply:
@taniguti 
If you have also any policy, please let us know.

Reply:
Almost cases today, this script will be used for upgrade Mojave. As you know in this case, auth reboot won't work as we expect. The latest Installer 10.14.3 will require reboot twice. This means the second reboot will request user to decrypt start up disk. So auth reboot feature won't end up saving end-user's time.

The other hands, this script supports earlier version of macOS update. like 10.10. 
If user uses this script for upgrade to 10.13, High Sierra, user may want to use auth reboot for their end user convenient. I think this script should keep that feature.

So I think it is important this script have a switch for that auth reboot feature enable. It should be depend on user's requirement. This is as same as idea 3 above.

Reply:
Thank you for your suggestions.
Then I think that there is also the following *Idea 4* as:

*Idea 4*. If the version of os installer is 10.14 or later and a non-administrative user runs this script, it should uniformly not auth reboot.

Reply:
idea 4 looks better.
I'd like to test it and make it as another PR.

Reply:
Thank you for your suggestion! 

Sounds good for me. Let’s go with it.
I will create an issue about this.

Reply:
I created https://github.com/kc9wwh/macOSUpgrade/issues/100.